feat: Synchronize content with GutenbergKit

dcalhoun opened the pull request:
Builds upon the in-progress Android integration to persist editor changes to the
host app, track editor changes from initial content, and populate the editor
with initial content for drafts (e.g., page templates).

Related:

* https://github.com/wordpress-mobile/GutenbergKit/pull/28

To test: See https://github.com/wordpress-mobile/GutenbergKit/pull/28.

## Regression Notes
1. Potential unintended areas of impact
    None, limited to the experimental editor.
2. What I did to test those areas of impact (or what existing automated tests I relied on)
    N/A, no targeted areas.
3. What automated tests I added (or what prevented me from doing so)
    N/A, not testing this experimental feature.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

dcalhoun committed:
feat: Persist GutenbergKit changes

Ensure post changes are not lost.

dcalhoun committed:
fix: Persist changes if either post title or content changes

The previous conditional erroneously ignored changes unless both values
changed.

wpmobilebot committed:
Merge release/25.4 into trunk (#23628)

* Bump version number

* Update draft release notes for 25.4.

* Update draft release notes for 25.4.

* Release Notes: add new section for next version (25.5)

* Update strings for localization

dcalhoun committed:
build: Update GutenbergKit ref

dangermattic commented:
<!--
  0 Errors
  2 Warnings: <code>Modules/Package.swift</c...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages in Xcode.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dcalhoun committed:
fix: Improve empty editor content check

As noted in the code comment, this value lags behind reality. A race
condition existed in production builds cause inaccurate values. This was
removed to instead rely upon the existing `editorHasContent`
implementation found in the `PostEditor` class.

dcalhoun committed:
build: Update GutenbergKit ref

dcalhoun committed:
build: Update GutenbergKit ref

dcalhoun committed:
build: Update GutenbergKit ref

dcalhoun commented on WordPress/Classes/ViewRelated/NewGutenberg/NewGutenbergViewController.swift:
```diff
@@ -76,11 +76,6 @@ class NewGutenbergViewController: UIViewController, PostEditor, PublishingEditor
         return !changes.isEmpty
     }
 
-    // TODO: this has to be incorrect and/or lagging behind
-    var editorHasContent: Bool {
-        !editorViewController.state.isEmpty
-    }
```
As noted in the comment, this was often incorrect. This was removed to rely upon the existing `editorHasContent` [implementation](https://github.com/wordpress-mobile/WordPress-iOS/blob/d3e49882603219af9756c8d41d381bf3cc01abe4/WordPress/Classes/ViewRelated/Post/PostEditor.swift#L77-L79), which appears to be more accurate.

Commit:
Update GutenbergKit ref

Commit:
Update GutenbergKit ref

geriux reviewed (approved):
LGTM! Nice work!

jkmassel committed:
Merge branch 'release/25.4' into feat/synchronize-content-with-gutebergkit
